src/db/oracledb/books-dao.js

```diff
@@ -60,6 +60,63 @@ const getBookById = async (id) => {
   }
 };
 
+/**
+ * Update a specific book by unique ID
+ *
+ * @param {string} id Unique book ID
+ * @param {object} updateData Data to update
+ * @param {object} existingBook Exisitng book to update
+ * @returns {Promise<object>} Promise object represents the updated book or undefined if not found
+ */
+const updateBookById = async (id, updateData, existingBook) => {
+  const connection = await getConnection();
+
+  try {
+    if (!existingBook) {
+      return undefined;
+    }
+
+    const updatedBookData = {
+      ...existingBook,
+      ...updateData,
+    };
+
+    const updateQuery = `
+      UPDATE library_api_books
+      SET title = :title,
+          author = :author,
+          publicationyear = :publicationYear,
+          isbn = :isbn,
+          genre = :genre,
+          description = :description,
+          available = :available
+      WHERE book_id = :bookId
+    `;
+
+    const bindVars = {
+      title: updatedBookData.title,
+      author: updatedBookData.author,
+      publicationYear: updatedBookData.publicationyear,
+      isbn: updatedBookData.isbn,
+      genre: updatedBookData.genre,
+      description: updatedBookData.description,
+      available: updatedBookData.available,
+      bookId: id,
+    };
+
+    const result = await connection.execute(updateQuery, bindVars);
+
+    if (result.rowsAffected === 1) {
+      await connection.commit();
+      return updatedBookData;
```

Comment:
I think a better way to confirm that your data is updated correctly is to fetch the data again after the update, instead of returning the data that you inserted. It's more accurate to reflect the data in real-time.

Reply:
Yes that I/O, network call can be expensive. But for now sure, I'll change to pulling out the inserted data and then returning it.

Reply:
Done, changed.